index.js

```diff
@@ -129,7 +205,16 @@ function filterByGenre() {}
       }
     ];
  */
-function getAllMoviesReleasedAtOrBeforeYear() {}
+function getAllMoviesReleasedAtOrBeforeYear(movies, year) {
+  let movieMatchYear = [];
+ 
+  for (let i = 0; i < movies.length; i++) {
+   if(Number((movies[i].released).slice(movies[i].released.length - 4, movies[i].released.length)) <= year) {
```

Comment:
Once again great logic.  For long chained operations like this consider breaking them apart on the invocation:
For long chain logic like this I'm a fan of using a variable to shorten and keep things clear.  
if we `let current = movies[i].released` we can shorten our call to :
`Number((current).slice(current.length - 4, current.length))`